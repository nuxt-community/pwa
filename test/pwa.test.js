jasmine.DEFAULT_TIMEOUT_INTERVAL = 30000
process.env.PORT = process.env.PORT || 5060
process.env.NODE_ENV = 'production'

const { Nuxt, Builder } = require('nuxt-edge')

const url = path => `http://localhost:${process.env.PORT}${path}`

describe('pwa', () => {
  let nuxt

  beforeAll(async () => {
    nuxt = new Nuxt(require('./fixture/nuxt.config'))
    await new Builder(nuxt).build()
    await nuxt.listen(process.env.PORT)
  })

  afterAll(async () => {
    await nuxt.close()
  })

  test('init', async () => {
    const window = await nuxt.renderAndGetWindow(url('/'))
    const headers = window.document.head.innerHTML
    expect(headers).toMatchSnapshot()
  })
})
