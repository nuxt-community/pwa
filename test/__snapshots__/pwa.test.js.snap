// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pwa init 1`] = `
"
    <title data-n-head=\\"true\\">Test Project Name</title><meta data-n-head=\\"true\\" data-hid=\\"charset\\" charset=\\"utf-8\\"><meta data-n-head=\\"true\\" data-hid=\\"viewport\\" name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\"><meta data-n-head=\\"true\\" data-hid=\\"mobile-web-app-capable\\" name=\\"mobile-web-app-capable\\" content=\\"yes\\"><meta data-n-head=\\"true\\" data-hid=\\"apple-mobile-web-app-title\\" name=\\"apple-mobile-web-app-title\\" content=\\"Test Project Name\\"><meta data-n-head=\\"true\\" data-hid=\\"description\\" name=\\"description\\" content=\\"Test Project Description\\"><meta data-n-head=\\"true\\" data-hid=\\"theme-color\\" name=\\"theme-color\\" content=\\"black\\"><meta data-n-head=\\"true\\" data-hid=\\"og:type\\" name=\\"og:type\\" property=\\"og:type\\" content=\\"website\\"><meta data-n-head=\\"true\\" data-hid=\\"og:title\\" name=\\"og:title\\" property=\\"og:title\\" content=\\"Test Project Name\\"><meta data-n-head=\\"true\\" data-hid=\\"og:site_name\\" name=\\"og:site_name\\" property=\\"og:site_name\\" content=\\"Test Project Name\\"><meta data-n-head=\\"true\\" data-hid=\\"og:description\\" name=\\"og:description\\" property=\\"og:description\\" content=\\"Test Project Description\\"><link data-n-head=\\"true\\" rel=\\"manifest\\" href=\\"/_nuxt/manifest.0fc4969c.json\\"><link data-n-head=\\"true\\" rel=\\"shortcut icon\\" href=\\"/_nuxt/icons/icon_64.9mgd2ZDMcQu.png\\"><link data-n-head=\\"true\\" rel=\\"apple-touch-icon\\" href=\\"/_nuxt/icons/icon_512.9mgd2ZDMcQu.png\\" sizes=\\"512x512\\"><script data-n-head=\\"true\\" async=\\"true\\" src=\\"/_nuxt/ons.40ac840e.js\\"></script><link rel=\\"preload\\" href=\\"/_nuxt/commons.app.js\\" as=\\"script\\"><link rel=\\"preload\\" href=\\"/_nuxt/vendor.css\\" as=\\"style\\"><link rel=\\"preload\\" href=\\"/_nuxt/app.js\\" as=\\"script\\"><link rel=\\"preload\\" href=\\"/_nuxt/pages/index.js\\" as=\\"script\\"><link rel=\\"stylesheet\\" href=\\"/_nuxt/vendor.css\\">
  "
`;
